### ISSUE TITLE
[agent_farm][difficult]

### ISSUE THREAD

**Issue by theskcd** [codestoryai/sidecar#1905]
I want to have a finish tool which provides data to a post completion script for sanity check..
I want to call this tool 'Wait...' this implies the LLM is going to think
It's like a backspace tool which the agent has access to, if it wants to course correct
We leave the option to the agentic model to invoke this tool and explore other options
We first try to get the model to use this tool, for which we want to create a new Tool called: Wait and allow the agent to take a step back and explore other options, we let the agentic LLM decide this and invoke it at will, we don't invoke it ourselves (ever, unless we get logits but thats another option)
we can also do this when we exceed a context window limit and let the agent do this

Take a first pass at building it,
I want the ToolType::Wait to be present
I want it to be part of the ToolInputPartial and have it registered with the tool/broker.rs as well

And then the ToolUseAgent should be registered with this tool as well
We can enable this when the reasoning is set to true in tool_use_agent_endpoint

**Comment by theskcd** [comment]
ACK, on it!

**Comment by theskcd** [comment]
I've pushed a new commit (3e2cc26) to this PR. Please review the changes.
